From 2104849fad6dcee58ebaea66020a329c0ca4649c Mon Sep 17 00:00:00 2001
From: Rubidium <rubidium@openttd.org>
Date: Tue, 17 Jan 2023 22:24:34 +0100
Subject: [PATCH 0845/1053] Fix #10368, Fix 994bf19: server restarting game
 caused clients to hit assertion

Upon closing the EndGameWindow, triggered from UnInitWindowSystem, the
HighScoreWindow would be opened and _z_windows would not be empty.
---
 src/highscore_gui.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/highscore_gui.cpp b/src/highscore_gui.cpp
index 9b57ba659a..f61bf46f37 100644
--- a/src/highscore_gui.cpp
+++ b/src/highscore_gui.cpp
@@ -126,7 +126,7 @@ struct EndGameWindow : EndGameHighScoreBaseWindow {
 	void Close() override
 	{
 		if (!_networking) Command<CMD_PAUSE>::Post(PM_PAUSED_NORMAL, false); // unpause
-		ShowHighscoreTable(this->window_number, this->rank);
+		if (_game_mode != GM_MENU) ShowHighscoreTable(this->window_number, this->rank);
 		this->EndGameHighScoreBaseWindow::Close();
 	}
 
-- 
2.55.0

